nat/src/main/java/org/hyperledger/besu/nat/kubernetes/service/ClusterIpBasedDetector.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.nat.kubernetes.service;
+
+import org.hyperledger.besu.nat.core.IpDetector;
+
+import java.util.Optional;
+
+import io.kubernetes.client.models.V1Service;
+
+public class ClusterIpBasedDetector implements IpDetector {
```

Comment:
Not worth creating a class for such a simple implementation IMO. You can inject the code as a lambda directly.

Reply:
Good catch